ide/projectapi/src/org/netbeans/spi/project/SingleMethod.java

```diff
@@ -43,14 +46,28 @@
      * @since 1.19
      */
     public SingleMethod(FileObject file, String methodName) {
-        super();
+        this(file, null, methodName);
+    }
+    
+    /**
+     * Creates a new instance holding the specified identification
+     * of a method/function in a file.
+     *
+     * @param file file to be kept in the object
+     * @param binaryClassName The binary class name of the class containing the method (as returned by {@link Class#getName()})
+     * @param methodName name of a method inside the file
+     * @exception  java.lang.IllegalArgumentException
+     *             if the file or method name is {@code null}
+     */
+    public SingleMethod(FileObject file, String binaryClassName, String methodName) {
```

Comment:
Changes in API should be reflected in `apichanges.xml`, version of the module should be incremented in `manifest.mf` or project properties. New elements should have `@since` tag. Plus: all modules that depend on the new elements should depend on the new version of the API module.

This is a request.

Reply:
Sure, will fix